src/save.ts

```diff
@@ -11,10 +11,18 @@ process.on("uncaughtException", e => utils.logWarning(e.message));
 
 async function run(): Promise<void> {
     try {
-        if (utils.isGhes()) {
-            utils.logWarning(
-                "Cache action is not supported on GHES. See https://github.com/actions/cache/issues/505 for more details"
-            );
+        if (!cache.isAvailable()) {
+            if (utils.isGhes()){
+                utils.logWarning(
+                    "Cache action is only supported on GHES version >= 3.5. If you are on version >=3.5 Please check with GHES admin if ArtifactCache service is enabled or not."
+                );
+            }
+            else{
+                utils.logWarning(
+                    "Something is going wrong with ArtifactCache service which supports cache actions. Please check https://www.githubstatus.com/ for any ongoing issue in actions."
+                );
+            }
```

Comment:
Looks like big enough repeated code and can think about moving to utils

Reply:
moved to function